lib/phoenix/verified_routes.ex

```diff
@@ -0,0 +1,607 @@
+defmodule Phoenix.VerifiedRoutes do
+  @moduledoc ~S"""
+  TODO
+    - [ ] ~p"/posts?page=#{page}"
+
+  use Phoenix.VerifiedRoutes,
+    router: AppWeb.Router,
+    endpoint: AppWeb.Endpoint,
+    statics: ~(images)
+  """
+  defstruct router: nil,
+            route: nil,
+            inspected_route: nil,
+            stacktrace: nil,
+            test_path: nil
+
+  defmacro __using__(opts) do
+    opts =
+      if Macro.quoted_literal?(opts) do
+        Macro.prewalk(opts, &expand_alias(&1, __CALLER__))
+      else
+        opts
+      end
+
+    quote do
+      Module.register_attribute(__MODULE__, :phoenix_verified_routes, accumulate: true)
+
+      opts = unquote(opts)
+
+      @before_compile unquote(__MODULE__)
+      @endpoint Keyword.get(opts, :endpoint)
+      @router Keyword.get(opts, :router)
+      @statics Keyword.get(opts, :statics, [])
+      import unquote(__MODULE__)
+    end
+  end
```

Comment:
```suggestion
    quote do
      unquote(__MODULE__).__using__(__MODULE__, unquote(opts))
      import unquote(__MODULE__)
    end
  end
  
  def __using__(mod, opts) do
    Module.register_attribute(mod, :phoenix_verified_routes, accumulate: true)
    Module.put_attribute(mod, :before_compile, __MODULE__)
    Module.put_attribute(mod, :router, Keyword.fetch!(opts, :router))
    Module.put_attribute(mod, :endpoint, Keyword.get(opts, :endpoint))
    Module.put_attribute(mod, :phoenix_verified_statics, Keyword.get(opts, :statics)
  end
```

Note I did two extra changes:

1. Made the :router required
2. Renamed :statics to :phoenix_verified_statics